src/Models/HydrostaticFreeSurfaceModels/update_hydrostatic_free_surface_model_state.jl

```diff
@@ -22,27 +22,26 @@ Update peripheral aspects of the model (auxiliary fields, halo regions, diffusiv
 hydrostatic pressure) to the current model state. If `callbacks` are provided (in an array),
 they are called in the end.
 """
-update_state!(model::HydrostaticFreeSurfaceModel, callbacks=[]) = update_state!(model, model.grid, callbacks)
+update_state!(model::HydrostaticFreeSurfaceModel, callbacks=[]; compute_tendencies = true) =
+         update_state!(model, model.grid, callbacks; compute_tendencies)
 
-function update_state!(model::HydrostaticFreeSurfaceModel, grid, callbacks)
+function update_state!(model::HydrostaticFreeSurfaceModel, grid, callbacks; compute_tendencies = true)
 
     @apply_regionally mask_immersed_model_fields!(model, grid)
 
-    fill_halo_regions!(prognostic_fields(model), model.clock, fields(model))
+    fill_halo_regions!(prognostic_fields(model), model.clock, fields(model); async = true)
 
     @apply_regionally replace_horizontal_velocity_halos!(model.velocities, model.grid)
     @apply_regionally compute_w_diffusivities_pressure!(model)
+    fill_halo_regions!(model.diffusivity_fields; only_local_halos = true)
 
-    fill_halo_regions!(model.velocities.w, model.clock, fields(model))
-    fill_halo_regions!(model.diffusivity_fields, model.clock, fields(model))
-    fill_halo_regions!(model.pressure.pHY′)
+    [callback(model) for callback in callbacks if isa(callback.callsite, UpdateStateCallsite)]
```

Comment:
```suggestion
    [callback(model) for callback in callbacks if callback.callsite isa UpdateStateCallsite]
```

why doesn't that work

Reply:
It should work